### PR TITLE
Added Api Endpoints

### PR DESCRIPTION
**Pull request by MaxTyrchan**
We need to test them (maybe together)!

### CODE REVIEW

File: src/types/applicant.ts [hunk 1]
```diff
@@ -14,3 +14,37 @@ export type Applicant = {
   skills: Skill[];
   rating: number;
 };
+
+export function createApplicant(applicants: Applicant) {
+  let result = [];
+  if (Array.isArray(applicants)) {
```
Comment (ScottPalme) [inline src/types/applicant.ts:20]: The if-check does not make sense, considering the input parameter is a single applicant

File: src/types/vacancy.ts [hunk 2]
```diff
@@ -12,3 +12,33 @@ export type Vacancy = {
   createdAt: string;
   updatedAt: string;
 };
+
+export function createVacancy(vacancies: Vacancy) {
+  let result = [];
+  if (Array.isArray(vacancies)) {
```
Comment (ScottPalme) [inline src/types/vacancy.ts:18]: Same things like with the applicant

File: src/api/api.tsx [hunk 3]
```diff
@@ -6,8 +9,12 @@ export default class API {
   // Local Python backend
   URL = 'http://127.0.0.1:5000/';
 
-  //Resume
-  addPdfsURL = () => `${this.URL}/upload`;
+  //URLs
+  addPdfURL = () => `${this.URL}/upload`;
+  getApplicantsURL = () => `${this.URL}/applicants`;
+  getApplicantURL = (id: string) => `${this.URL}/applicants/${id}`;
```
Comment (ScottPalme) [inline src/api/api.tsx:15]: Routes which handle single objects should be named in singular

File: src/api/api.tsx [hunk 4]
```diff
@@ -35,10 +42,48 @@ export default class API {
...
-      body: pdfs,
+      body: pdf,
+    });
+  }
+
+  getApplicants() {
+    return this.fetchAdvanced(this.getApplicantsURL(), {
+      method: 'GET',
+    }).then((res) => {
+      let applicants = createApplicant(res);
```
Comment (ScottPalme) [inline src/api/api.tsx:56]: The responses should be named properly with the correct type:
Example:
- `res` should be called `applicant` with the type `Applicant`

**Review by ScottPalme — CHANGES_REQUESTED** [review]
Have a look at the suggestions

File: src/types/applicant.ts (hunk 1 above)
Comment (MaxTyrchan) [reply src/types/applicant.ts:20]: That is why there's an if clause... to be able to convert both (a single applicant and multiple applicants) into the right format. I would leave it like that but i can also split it up into two function.

File: src/api/api.tsx (hunk 4 above)
Comment (MaxTyrchan) [reply src/api/api.tsx:56]: Have you considered that we not receiving one applicant but many applicants do to the name of the function?
Therefore the name 'applicant' for the variable 'res' would be 'applicants' which doubles with the variable name in line 56. In addition the type of 'res' is not Applicant but a json string.

File: src/api/api.tsx (hunk 3 above)
Comment (MaxTyrchan) [reply src/api/api.tsx:15]: Fixed it

**Comment by MaxTyrchan** [comment]
I updated some of the files and commited them but didn't resolve the conflicts because i thought we agreed on that the reviewer should do this. Hope that was correct :)

**Review by ScottPalme — APPROVED** [review]

**Review by ScottPalme — APPROVED** [review]

**Pull request merged**